src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,187 @@
+package parser;
+
+import exception.DukeException;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import task.Constant;
+
+public class Parser {
+    /**
+     * @param input raw input
+     * @return Boolean returns true if input is a done/delete command
+     * @throws DukeException if user input matches a done/delete command but it not properly
+     *     formatted
+     */
+    private static String[] timeRegex = {
+        "H:m", "Hmm", "H.m", "h.m a", "h.m a", "h:m a", "h:m a", "hmm a", "hmma", "h a", "ha", "H"
+    };
+
+    private static String[] dateRegex = {
+        "ddMMyyyy", "yyyyMMdd", "d-M-yyyy", "d/M/yyyy", "yyyy-M-d", "yyyy/M/d"
+    };
+
+    public static Boolean isDoneDelete(String input) throws DukeException {
+        if (Pattern.matches(
+                "(^(done|delete)\\s+.*|(.*\\s+(done|delete)\\s+.*)|.*\\s+(done|delete)$)", input)) {
+            if (!Pattern.matches("^(done|delete)\\s+.*", input)) {
+                throw new DukeException("Action should be at the front");
+            }
+            if (Pattern.matches("^(done|delete)\\s+\\d+\\D+$", input)) {
+                throw new DukeException("Must end with a number and provide only one number!");
+            }
+            if (!Pattern.matches("^(done|delete)\\s+\\d+$", input)) {
+                throw new DukeException("A task number must be provided");
+            }
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * @param input raw user input
+     * @return Boolean returns true if command is a find command
+     * @throws DukeException if user input matches a find command but it not properly formatted
+     */
+    public static Boolean isFind(String input) throws DukeException {
+        if (Pattern.matches("^(find\\s+.*)|(.*\\s+find\\s+.*)|(.*\\s+find$)", input)) {
+            if (!Pattern.matches("^find\\s+.*", input)) {
+                throw new DukeException("Action should be at the front");
+            }
+            if (!Pattern.matches("^find\\s+.*", input)) {
+                throw new DukeException("A task number must be provided");
+            }
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * @param input raw input
+     * @return String the lower cased task type is returned (todo/deadline/event)
+     * @throws DukeException when type is not recognized
+     */
+    public static String getType(String input) throws DukeException {
+        String lowerCaseInput = input.toLowerCase();
+        String acceptedTypes = String.format("(%s)", String.join("|", Constant.taskTypes));
+        if (Pattern.matches(
+                String.format(
+                        "^%s\\s+.*|.*\\s+%s$|.*\\s+%s\\s+.*",
```

Comment:
Good job on making the regexes. One important suggestion is that while using such regexes, it is highly recommended to add a comment explaining the regex and someone else might need to debug your regex and it is not easy to understand it by just reading it.